lib/cloudregister/registerutils.py

```diff
@@ -2668,6 +2663,20 @@ def __remove_service(smt_server_name):
     return 1
 
 
+# ----------------------------------------------------------------------------
+def __remove_state_file(filepath):
+    """Remove the given file if it does not exist"""
+    status = True
+    if os.path.exists(filepath):
+        status = False
+        try:
+            os.unlink(filepath)
+            status = True
+        except OSError as error:
+            logging.error(error)
+    return status
+
```

Comment:
If another exception than an OSError appears, the method will raise and the application probably exits with a stacktrace. Is this behavior wanted/expected ? I'm asking because other state cleanups could be skipped in case another than an OSError exception is raised.

In case we want to capture all exceptions. This can be changed to

```python3
except Exception as error:
    logging.error('State cleanup for {0} failed with {1}'.format(filepath, error))
```
```

Reply:
Done, skipped the OS error treating all errors equally